client/signup/jetpack-connect/index.jsx

```diff
@@ -174,16 +171,18 @@ const JetpackConnectMain = React.createClass( {
 		}
 
 		if ( this.state.waitingForSites && ! this.props.isRequestingSites ) {
+			// FIXME: Can this be improved?
```

Comment:
@tyxla I just spent a very long time looking at this. Here's my assessment:

`state.waitingForSites` does not solely depend on `props.isRequestingSites`. When `onURLEnter` is called, if `props.isRequestingSites` then it sets `waitingForSites` and defers a call to `checkUrl`, otherwise it makes the call immediately. Then, when `isRequestingSites` changes and we enter the `componentDidUpdate` call, we can hit the condition we're commenting and make the `checkUrl` call.

If we were to eliminate `waitingForSites`, we couldn't do this wait-then-check flow.

Why are we waiting on `isRequestingSites`? Is that just to lighten network load? There doesn't seem to be any reason the two are related. It's not clear to me now that `isRequestingSites` has _any_ relevance for this component.